src/psyclone/tests/domain/lfric/lfric_builtins_test.py

```diff
@@ -85,17 +86,109 @@ def dummy_func(self, _1, _2=True):
     self._proxy_data_type = "dummy3"
     self._module_name = "dummy4"
 
+
+class Dummy(LFRicBuiltIn):
+    ''' Utility subclass to allow abstract LFRicBuiltin class to be tested '''
+    @staticmethod
+    def metadata():
+        ''' Subclass abstract method.
+
+        :returns: None
+        :rtype: NoneType
+
+        '''
+        return None
+
+
+class Dummy2(Dummy):
+    '''Utility subclass to allow abstract LFRicBuiltin class
+    _builtin_metadata method to be tested'''
+    _case_name = "test"
+
+
+class XKernDummy(LFRicXKern):
+    '''Utility class to test that LFRicXKern raises the expected
+    exception
+
+    '''
+    @staticmethod
+    def metadata():
+        return None
+
+    def __str__(self):
+        return "dummy"
+
 # ------------- Tests for built-ins methods and arguments ------------------- #
 
 
-def test_lfric_builtin_abstract_methods():
-    ''' Check that the LFRicBuiltIn class is abstract and that the __str__
+def test_lfric_builtin_abstract_method():
+    ''' Check that the LFRicBuiltIn class is abstract and that the metadata
     method is abstract. '''
     with pytest.raises(TypeError) as err:
         # pylint: disable=abstract-class-instantiated
         lfric_builtins.LFRicBuiltIn()
     assert "abstract class LFRicBuiltIn" in str(err.value)
-    assert "__str__" in str(err.value)
+    assert "metadata" in str(err.value)
+
+
+def test_lfric_builtin_init():
+    '''Check initiaisation of the abstract LFRicBuiltIn class.'''
+    instance = Dummy()
+    # Check __init__
+    assert instance.qr_rules == {}
```

Comment:
This is one of the pylint warnings we ignore for tests. If you run pylint from the tests directory you should find that you get 10/10. `cd psyclone/src/psyclone/tests; pylint domain/lfric/lfric_builtins_test.py`. The reason I use this particular check here is because I really want to ensure that an empty dictionary is returned. It would be wrong if `False`, or `None`, or 0 were returned. But these values would pass the test if we changed to `not instance.qr_rules`.

Reply:
Thanks for the explanation - confirmed 10/10!